pallets/clusters/src/lib.rs

```diff
@@ -60,10 +58,8 @@ pub struct Cluster<TAccountId> {
 	pub name: Vec<u8>,
 	/// The ID of the cluster
 	pub cluster_id: Hash128,
-	/// The list of Roles associated to the Cluster
-	pub roles: Vec<Role<TAccountId>>,
-	/// The list of Members of the Cluster
-	pub members: Vec<TAccountId>,
+	/// The map that contains the list of Roles in Cluster, each associated with an index
+	pub roles: BTreeMap<u64, Role>,
```

Comment:
index is confusing, that's actually the ID of the name no?

Reply:
yeah, actually another index is not be needed at all, I realize. This could be a `BTreeMap<Vec<u8>, Role>` with `Vec<u8>` being simply the role name.

Reply:
👎 we want to cache names like tags in protos actually. We don't want repeating names. Check protos metadata. What I meant was more regarding the fact it was defined index in your comments, it's more an ID of the name instead (like in protos, if we called it index in protos it's also not correct btw)

Reply:
Let's have a meeting about this. Let's avoid too much back and forth again.